pkg/cmd/issue/edit/edit.go

```diff
@@ -157,59 +173,98 @@ func editRun(opts *EditOptions) error {
 		lookupFields = append(lookupFields, "milestone")
 	}
 
-	issue, repo, err := shared.IssueFromArgWithFields(httpClient, opts.BaseRepo, opts.SelectorArg, lookupFields)
+	// Get all specified issues and make sure they are within the same repo.
+	issues, repo, err := shared.IssuesFromArgsWithFields(httpClient, opts.BaseRepo, opts.SelectorArgs, lookupFields)
 	if err != nil {
 		return err
 	}
 
-	editable.Title.Default = issue.Title
-	editable.Body.Default = issue.Body
-	editable.Assignees.Default = issue.Assignees.Logins()
-	editable.Labels.Default = issue.Labels.Names()
-	editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
-	projectItems := map[string]string{}
-	for _, n := range issue.ProjectItems.Nodes {
-		projectItems[n.Project.ID] = n.ID
-	}
-	editable.Projects.ProjectItems = projectItems
-	if issue.Milestone != nil {
-		editable.Milestone.Default = issue.Milestone.Title
-	}
-
-	if opts.Interactive {
-		err = opts.FieldsToEditSurvey(&editable)
-		if err != nil {
-			return err
-		}
-	}
-
+	// Fetch editable shared fields once for all issues.
 	apiClient := api.NewClientFromHTTP(httpClient)
-	opts.IO.StartProgressIndicator()
+	opts.IO.StartProgressIndicatorWithLabel("Fetching repository information")
 	err = opts.FetchOptions(apiClient, repo, &editable)
 	opts.IO.StopProgressIndicator()
 	if err != nil {
 		return err
 	}
 
-	if opts.Interactive {
-		editorCommand, err := opts.DetermineEditor()
-		if err != nil {
-			return err
-		}
-		err = opts.EditFieldsSurvey(&editable, editorCommand)
-		if err != nil {
-			return err
-		}
+	// Update all issues in parallel.
+	if opts.Workers == 0 {
+		opts.Workers = 10
 	}
+	g, ctx := errgroup.WithContext(context.Background())
+	g.SetLimit(opts.Workers)
+
+	issueURLs := make([]string, 0, len(issues))
+	issuesChan := make(chan string)
+	doneCh := make(chan any)
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				doneCh <- nil
+				return
+			case issueURL, ok := <-issuesChan:
+				if ok {
+					issueURLs = append(issueURLs, issueURL)
+				}
+			}
+		}
+	}()
+
+	opts.IO.StartProgressIndicatorWithLabel(fmt.Sprintf("Updating %d issues", len(issues)))
+	for _, issue := range issues {
+		issue := *issue
+		g.Go(func() error {
+			editable.Title.Default = issue.Title
+			editable.Body.Default = issue.Body
+			editable.Assignees.Default = issue.Assignees.Logins()
+			editable.Labels.Default = issue.Labels.Names()
+			editable.Projects.Default = append(issue.ProjectCards.ProjectNames(), issue.ProjectItems.ProjectTitles()...)
+			projectItems := map[string]string{}
+			for _, n := range issue.ProjectItems.Nodes {
+				projectItems[n.Project.ID] = n.ID
+			}
+			editable.Projects.ProjectItems = projectItems
+			if issue.Milestone != nil {
+				editable.Milestone.Default = issue.Milestone.Title
+			}
```

Comment:
Using `editable` across the Go routines seems like an issue... I would think it would be safer to create a new `prShared.Editable` for each issue that needs editing.